lib/ansible/modules/cloud/vmware/vmware_drs_group.py

```diff
@@ -0,0 +1,600 @@
+        if state == 'present':
+
+            if self.__vm_list is not None:
+                self.__set_vm_obj_list(vm_list=self.__vm_list)
+
+            if self.__host_list is not None:
+                self.__set_host_obj_list(host_list=self.__host_list)
+
+    def get_msg(self):
+        """
+        Returns message for Ansible result
+        Args: none
+
+        Returns: string
+        """
+        return self.__msg
+
+    def get_result(self):
+        """
+        Returns result for Ansible
+        Args: none
+
+        Returns: dict
+        """
+        return self.__result
+
+    def __set_result(self, group_obj):
+        """
+        Creates result for successfull run
+        Args:
+            group_obj: group pbject
+
+        Returns: None
+
+        """
+        self.__result = dict()
+
+        if (self.__cluster_obj is not None) and (group_obj is not None):
+            self.__result[self.__cluster_obj.name] = []
+            self.__result[self.__cluster_obj.name].append(self.__normalize_group_data(group_obj))
+
+    def get_changed(self):
+        """
+        Returns if anything changed
+        Args: none
+
+        Returns: boolean
+        """
+        return self.__changed
+
+    def __set_vm_obj_list(self, vm_list=None, cluster_obj=None):
+        """
+        Function pupulate vm object list from list of vms
+        Args:
+            vm_list: List of vm names
+
+        Returns: None
+
+        """
+
+        if vm_list is None:
+            vm_list = self.__vm_list
+
+        if cluster_obj is None:
+            cluster_obj = self.__cluster_obj
+
+        if vm_list is not None:
+
+            for vm in vm_list:
+
+                if self.module.check_mode is False:
+
+                    # Get host data
+                    vm_obj = find_vm_by_id(content=self.content, vm_id=vm,
+                                           vm_id_type='vm_name', cluster=cluster_obj)
+
+                    if vm_obj is None:
+                        raise Exception("VM %s does not exist in cluster %s" % (vm,
+                                                                                self.__cluster_name))
+
+                    self.__vm_obj_list.append(vm_obj)
+
+    def __set_host_obj_list(self, host_list=None):
+        """
+        Function pupulate host object list from list of hostnames
+        Args:
+            host_list: List of host names
+
+        Returns: None
+
+        """
+
+        if host_list is None:
+            host_list = self.__host_list
+
+        if host_list is not None:
+
+            for host in host_list:
+
+                if self.module.check_mode is False:
+
+                    # Get host data
+                    host_obj = self.find_hostsystem_by_name(host)
+
+                    if host_obj is None and self.module.check_mode is False:
+                        raise Exception("ESXi host %s does not exist in cluster %s" % (host, self.__cluster_name))
```

Comment:
Same as above, please fail the module, instead of raising an exception
```suggestion
            self.module.fail_json(msg="ESXi host %s does not exist in cluster %s" % (host, self.__cluster_name))
```